src/testing/mocks/middleware/icache.ts

```diff
@@ -0,0 +1,59 @@
+/* tslint:disable:interface-name */
+import { create, invalidator } from '../../../core/vdom';
+import { MiddlewareResult } from '../../../core/interfaces';
+import { cache } from '../../../core/middleware/cache';
+import { icache } from '../../../core/middleware/icache';
+import Map from '../../../shim/Map';
+
+export function createICacheMock() {
+	const map = new Map<string, any>();
+	const factory = create({ cache, invalidator });
+	const mockICacheFactory = factory(({ id, middleware, properties, children }) => {
+		const { callback } = icache();
+		const icacheMiddleware = callback({
+			id,
+			middleware: { invalidator: middleware.invalidator, cache: middleware.cache },
+			properties,
+			children
+		});
+		const setter = icacheMiddleware.set;
+
+		icacheMiddleware.set = (key: any, value: any) => {
+			if (typeof value === 'function') {
```

Comment:
I think we need to call the original setter with the value if it's not a function too.

Reply:
Definitely.